src/Microsoft.DotNet.Build.Tasks/Targets/partialfacades.targets

```diff
@@ -0,0 +1,141 @@
+    </PropertyGroup>
+
+    <Error 
+      Condition="!Exists('$(CustomPartialFacadeJsonFile)')"
+      Text="Partial facades require a project.json containing their contract package under src\partial-facade-packages\project.json."
+      />
+
+    <!-- Restore the special partial facade json file, which should contain the contract package to use in GenFacades -->
+    <Exec Command="$(DnuRestoreCommand) &quot;$(CustomPartialFacadeJsonFile)&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
+
+      <!-- Use new task for project.json -->
+    <ResolveNuGetPackageAssets Condition="'$(ProjectJson)' != ''"
+                               AdditonalProjectJsonFiles="$(CustomPartialFacadeJsonFile)"
+                               Architecture="$(PlatformTarget)"
+                               Configuration="$(Configuration)"
+                               Language="$(Language)"
+                               IngoreLockFile="$(NuGetIngoreLockFile)"
+                               PackageRoot="$(PackagesDir)"
+                               ProjectFile="$(MSBuildProjectFullPath)"
+                               TargetFrameworkMonikers="$(NugetTargetFrameworkMoniker)"
+                               TargetPlatformMonikers="$(TargetPlatformMoniker)"
+                               TransitiveProjectReferences="">
+      <Output TaskParameter="ResolvedReferences" ItemName="PossibleContractRefs" />
+    </ResolveNuGetPackageAssets>
+
+    <ItemGroup>
+      <ResolvedMatchingContract Include="@(PossibleContractRefs)" Condition="'%(FileName)' == '$(AssemblyName)'" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- FillPartialFacade
+       Inputs:
+         * An "input assembly"
+         * A contract assembly
+         * Seed assemblies
+
+       Fills the "input assembly" by finding types in the contract assembly that are missing from it, and adding type-forwards
+         to those matching types in the seed assemblies. -->
+
+  <!-- Inputs and outputs of FillPartialFacade -->
+  <PropertyGroup Condition="'$(IsPartialFacadeAssembly)' == 'true'">
+    <PartialFacadeAssemblyPath>$(IntermediateOutputPath)$(TargetName)$(TargetExt)</PartialFacadeAssemblyPath>
+    <PartialFacadeSymbols>$(IntermediateOutputPath)$(TargetName).pdb</PartialFacadeSymbols>
+    <GenFacadesInputPath>$(IntermediateOutputPath)PreGenFacades\</GenFacadesInputPath>
+    <GenFacadesInputAssembly>$(GenFacadesInputPath)$(TargetName)$(TargetExt)</GenFacadesInputAssembly>
+    <GenFacadesInputSymbols>$(GenFacadesInputPath)$(TargetName).pdb</GenFacadesInputSymbols>
+    <GenFacadesOutputPath>$(IntermediateOutputPath)</GenFacadesOutputPath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PostProcessAssemblySteps Include="FillPartialFacade" />
+  </ItemGroup>
+
+  <Target Name="FillPartialFacade"
+      Condition="'$(IsPartialFacadeAssembly)' == 'true'"
+      BeforeTargets="OpenSourceSign">
```

Comment:
Given the way you are doing CallTargets I'm not sure that this BeforeTargets actually does what you want.